Add Modal metrics data to integration

pawalt opened the pull request:
### What does this PR do?

This PR updates the documentation for the Modal integration to include information about metrics, and it adds the reported metrics to metadata.csv.

### Motivation

We are building out the Modal metrics integration for Datadog, and we're listing out our metrics so you know which metrics to expect.

### Review checklist

- [x] PR has a [meaningful title](https://github.com/DataDog/integrations-core/blob/master/CONTRIBUTING.md#pull-request-title) or PR has the `no-changelog` label attached
- [x] Git history is clean
- [x] If PR impacts documentation, docs team has been notified or an issue has been opened on the [documentation repo](https://github.com/DataDog/documentation/issues/new)

### Additional Notes

Thanks for the review!

eho1307 commented on modal/CHANGELOG.md:
```diff
@@ -1,5 +1,11 @@
+
+***Added***:
+
+* Modal now publishes container metrics to Datadog.
```
```suggestion
* Container metrics
```

pawalt replied:
done

eho1307 commented on modal/README.md:
```diff
@@ -37,7 +37,7 @@ Modal does not include any events.
 ### Metrics
 
-There are no metrics collected for this integration.
+Modal publishes container metrics to Datadog.
```
```suggestion
See [metadata.csv][3] for a list of metrics included with this integration.
```

pawalt replied:
done

eho1307 reviewed (changes requested):
A few points of feedback:
1. Could you include these metrics in your dashboard or create a new dashboard to include these metrics? Our dashboards help customers get started quickly. You can follow these [docs](https://docs.datadoghq.com/developers/integrations/create-an-integration-dashboard?tab=ui).
2. Please add the [metrics section](https://docs.datadoghq.com/developers/integrations/check_references/#manifest-file) within the manifest.json. View an example [here](https://github.com/DataDog/integrations-extras/blob/master/ably/manifest.json).

hestonhoffman commented on modal/README.md:
On line 65, please add the following:
```
[5]: https://github.com/DataDog/integrations-extras/blob/master/modal/metadata.csv
```

pawalt replied:
Nice thanks

pawalt commented:
@eho1307 @hestonhoffman thanks for the feedback - I've added the appropriate metadata, and I've added a new dashboard with a new screenshot. Let me know if there's anything else I can fix up.

eho1307 commented:
Looks good from my perspective. Once we have everyone's approval, we'll merge the PR.

pawalt commented:
thanks @eho1307!

pawalt commented:
@hestonhoffman let me know if there's anything I can do to improve this PR - I'm happy to make any changes that would be helpful :)

janine-c commented on modal/README.md:
```diff
@@ -37,7 +37,7 @@ Modal does not include any events.
 ### Metrics
 
-There are no metrics collected for this integration.
+See [metadata.csv][5] for a list of metrics included with this integration.
```
Not nitpicking your writing here - I think we need to use exact wording to ensure the "magic phrase" works properly:
```suggestion
See [metadata.csv][5] for a list of metrics provided by this integration.
```

janine-c reviewed (changes requested):
Hi, I'm stepping in for Heston while he's OOO. Just made a small suggestion request to ensure the "magic phrase" can really do its magic :)

pawalt commented:
@janine-c thanks! i've made this change

pawalt commented:
I think we just need ecosystems here - is there someone there who can take a look? Maybe @bgoldberg122 or @james-eichelbaum?

eho1307 commented:
Hi @pawalt we'll be releasing this tomorrow. No more reviews are needed.

pawalt commented:
@eho1307 thanks so much!
